### ISSUE TITLE
Can't seem to add new devices

### ISSUE THREAD

**Issue by rockymountainnative** [home-assistant/home-assistant.github.io#18612]
### Feedback

I don't know if the auto discover works. I have incrementally added thermostats to my house room by room and each time I add a new one, the Lyric intergration doesn't discover it. The only workaround I know of is to delete the integration and re-add it to where I am prompted through the honeywell api UI interaction where I can select the additional device that needs to be shared. Is there an easier way to accomplish this?

The honeywell feature I am looking for appears to be "Select Devices"

### URL

https://www.home-assistant.io/integrations/lyric/

### Version

2021.7.4

**Comment by frenck** [comment]
Devices are loaded from the Lyric API when the integration is set up. This happens when adding it, reloading the integration instance or when Home Assistant restarts.

**Comment by rockymountainnative** [comment]
Strange, I still do not seeing this automatically seeing new thermostats as I add them one by one each time. While the integration reloads, I do not think it goes through the setup process to try to permission my new devices to my api applicaiton.

The workaround I found was to save off the url that does the authentication: https://api.honeywell.com/oauth2/app/login?apikey=XXXXXX&redirect_uri=http%3A%2F%2Fhome-assistant.local%3A8123%2Fauth%2Fexternal%2Fcallback&app=Home%20Assistant&state=XXXXXXXX&lnsReg=true&subSystemId=1&futureDevices=null&source=XXXXXX
(any personal ids have been scrubbed)

That takes me to the allow page:

![image](https://user-images.githubusercontent.com/17169682/128107790-2663252a-3c99-4989-9cb2-3e180d1b75b9.png)

Select "Allow"

From there, I can opt in to any new devices that have been added via my phone app:

![image](https://user-images.githubusercontent.com/17169682/128107872-b1870a8e-aa34-4735-ae39-97870969d505.png)


From here, I reloaded the integration and everything showed up. The hard part here is that this is my only known work around without removing the integration, losing my custom names/entity_ids, and then re-adding the Lyric integration.

At any rate, many thanks for this integration, it is coming in very useful!
